Tests/SwiftNavigationTests/UIBindableTests.swift

```diff
@@ -9,8 +9,7 @@ private final class Model {
 }
 
 final class UIBindableTests: XCTestCase {
-  @MainActor
-  func testDynamicMemberLookupBindable() throws {
+  func testDynamicMemberLookupBindable() {
```

Comment:
Some tests were marked as `@MainActor` that didn't need it, so removing.

Also I moved this test from UIKitNavigation to SwiftNavigation since this is a pure Swift test.